experimental/fake/src/fake.ts

```diff
@@ -273,3 +273,99 @@ export function fake(
 
   console.warn(`Unsupported type: ${type}`)
 }
+export function empty(
+  spec: z.ZodTypeAny,
+  options: Partial<FakeOptions> = defaultOptions,
+): any {
+  const type: string = spec._def.typeName
+
+  if (type === 'ZodString') {
+    return ''
+  }
+
+  if (type === 'ZodNumber') {
+    return 0
+  }
+
+  if (type === 'ZodBoolean') {
+    return false
+  }
+
+  if (type === 'ZodDate') {
+    return faker.date.past()
+  }
+
+  if (type === 'ZodLiteral') {
+    return ''
+  }
+
+  if (type === 'ZodEnum') {
+    return spec._def.values[0]
+  }
+
+  if (type === 'ZodArray') {
+    return []
+  }
+
+  if (type === 'ZodTuple') {
+    return spec._def.items.map((s: z.ZodTypeAny) => empty(s, options))
+  }
+
+  if (type === 'ZodObject') {
+    const shape = spec._def.shape()
+    return Object.entries(shape).reduce((acc: any, [key]: any) => {
+      acc[key] = empty(shape[key], options)
+      return acc
+    }, {})
+  }
+
+  if (type === 'ZodRecord') {
+    return {}
+  }
+
+  if (type === 'ZodMap') {
+    return new Map()
+  }
+
+  if (type === 'ZodSet') {
+    return new Set()
+  }
+
+  if (type === 'ZodUnion') {
+    const types = spec._def.options
+    const one = faker.number.int({ min: 0, max: types.length - 1 })
```

Comment:
Avoid faker in empty. empty should be deterministic where possible. Use 0th element.